substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/SubstrateJVM.java

```diff
@@ -206,25 +226,29 @@ public boolean destroyJFR() {
 
         globalMemory.teardown();
         symbolRepo.teardown();
+        threadRepo.teardown();
+        stackTraceRepo.teardown();
+        methodRepo.teardown();
 
         initialized = false;
         return true;
     }
 
     /** See {@link JVM#getStackTraceId}. */
-    public long getStackTraceId(@SuppressWarnings("unused") int skipCount) {
-        // Stack traces are not supported at the moment.
-        return 0;
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+    public long getStackTraceId(int skipCount) {
+        return stackTraceRepo.getStackTraceId(skipCount, false);
     }
 
     /** See {@link JVM#getThreadId}. */
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
```

Comment:
There is no need for this method to be uninterruptible as it is not necessarily safe to call from uninterruptible code (see my other comments).

Reply:
Done.